services/vmagent-config-updater/models/config.go

```diff
@@ -0,0 +1,166 @@
+}
+
+// ScrapeConfig represents essential parts for `scrape_config` section of Prometheus config.
+//
+// See https://prometheus.io/docs/prometheus/latest/configuration/configuration/#scrape_config
+type ScrapeConfig struct {
+	JobName        string         `yaml:"job_name"`
+	ScrapeInterval time.Duration  `yaml:"scrape_interval,omitempty"`
+	ScrapeTimeout  time.Duration  `yaml:"scrape_timeout,omitempty"`
+	StaticConfigs  []StaticConfig `yaml:"static_configs,omitempty"`
+}
+
+// StaticConfig represents essential parts for `static_config` section of Prometheus config.
+//
+// See https://prometheus.io/docs/prometheus/latest/configuration/configuration/#static_config
+type StaticConfig struct {
+	Targets []string          `yaml:"targets"`
+	Labels  map[string]string `yaml:"labels,omitempty"`
+}
+
+func WithScrapeInterval(scrapeInterval time.Duration) ConfigOptions {
+	return func(config *Config) {
+		config.Global = GlobalConfig{ScrapeInterval: scrapeInterval}
+	}
+}
+
+func WithTargetCount(targetCount int) ConfigOptions {
+	return func(config *Config) {
+		if targetCount == 0 {
+			targetCount = defaultTargetCount
+			logger.Infof("targetCount must be greater than 0. Used default value")
+		}
+		config.targetCount = targetCount
+	}
+}
+
+func WithTargetsToUpdatePercentage(targetsToUpdatePercentage int) ConfigOptions {
+	return func(config *Config) {
+		if targetsToUpdatePercentage > 100 || targetsToUpdatePercentage < 0 {
+			targetsToUpdatePercentage = defaultTargetsToUpdatePercentage
+			logger.Infof("targetsToUpdatePercentage can't be lower than 0 or more than 100. Used default value")
+		}
+		config.targetsToUpdatePercentage = targetsToUpdatePercentage
+	}
+}
+
+func WithTargetAddr(targetAddr string) ConfigOptions {
+	return func(config *Config) {
+		host, port, err := net.SplitHostPort(targetAddr)
+		if err != nil {
+			targetAddr = defaultTargetAddr
+			logger.Infof("targetName not correct. Used default value")
+		} else if host == "" || port == "" {
+			targetAddr = defaultTargetAddr
+			logger.Infof("targetName can't be empty or must include correct host:port configuration. Used default value")
+		}
+		config.targetAddr = targetAddr
+	}
+}
+
+func WithJobName(jobName string) ConfigOptions {
+	return func(config *Config) {
+		if jobName == "" {
+			jobName = defaultJobName
+		}
+		config.jobName = jobName
+	}
+}
+
+func (cfg *Config) prepareStaticConfig() {
+	for i := 0; i < cfg.targetCount; i++ {
+		cfg.stConfigs = append(cfg.stConfigs, StaticConfig{
+			Targets: []string{cfg.targetAddr},
+			Labels: map[string]string{
+				"host_number": fmt.Sprintf("cfg_%d", i),
+				"instance":    strconv.FormatInt(time.Now().UnixNano(), 10),
+			},
+		})
+	}
+}
+
+func (cfg *Config) update() {
+	if len(cfg.stConfigs) == cfg.targetCount {
+		num := float64(cfg.targetCount) * (float64(cfg.targetsToUpdatePercentage) / 100)
+		for i := 0; i < int(num); i++ {
+			cfg.stConfigs[i] = StaticConfig{
```

Comment:
I meant the following by "shuffling":

```suggestion
		for i := 0; i < int(num); i++ {
		         j := rand.Intn(len(cfg.stConfigs))
			cfg.stConfigs[j] = StaticConfig{
```
Don't forget to seed the rand